how-to/common/public/views/theme/theme-definition-builder/theme-definition-builder-view.js

```diff
@@ -0,0 +1,508 @@
+
+// theme options
+const title = document.querySelector('#title');
+const logo = document.querySelector('#logo');
+
+const brandPrimary = document.querySelector('#brand-primary');
+const brandSecondary = document.querySelector('#brand-secondary');
+const backgroundPrimary = document.querySelector('#background-primary');
+const background1 = document.querySelector('#background-1');
+const background2 = document.querySelector('#background-2');
+const background3 = document.querySelector('#background-3');
+const background4 = document.querySelector('#background-4');
+const background5 = document.querySelector('#background-5');
+const background6 = document.querySelector('#background-6');
+const statusSuccess = document.querySelector('#status-success');
+const statusWarning = document.querySelector('#status-warning');
+const statusCritical = document.querySelector('#status-critical');
+const statusActive = document.querySelector('#status-active');
+const inputBackground = document.querySelector('#input-background');
+const inputColor = document.querySelector('#input-color');
+const inputPlaceholder = document.querySelector('#input-placeholder');
+const inputDisabled = document.querySelector('#input-disabled');
+const inputFocused = document.querySelector('#input-focused');
+const textDefault = document.querySelector('#text-default');
+const textHelp = document.querySelector('#text-help');
+const textInactive = document.querySelector('#text-inactive');
+
+// variables
+let lastDomain;
+let lastColor;
+const paletteLight = {
+	brandPrimary: '#399BB8',
+	brandSecondary: '#383A40',
+	backgroundPrimary: '#111214',
+	background1: '#ECEEF1',
+	background2: '#FAFBFE',
+	background3: '#ECEEF1',
+	background4: '#ECEEF1',
+	background5: '#DDDFE4',
+	background6: '#ffffff',
+	statusSuccess: '#35C759',
+	statusWarning: '#C93400',
+	statusCritical: '#000000',
+	statusActive: '#399BB8',
+	inputBackground: '#ECEEF1',
+	inputColor: '#1E1F23',
+	inputPlaceholder: '#7D808A',
+	inputDisabled: '#7D808A',
+	inputFocused: '#C9CBD2',
+	textDefault: '#1E1F23',
+	textHelp: '#111214',
+	textInactive: '#7D808A'
+};
+
+const paletteDark = {
+	brandPrimary: '#504CFF',
+	brandSecondary: '#383A40',
+	backgroundPrimary: '#000000',
+	background1: '#111214',
+	background2: '#1E1F23',
+	background3: '#24262B',
+	background4: '#2F3136',
+	background5: '#383A40',
+	background6: '#53565F',
+	statusSuccess: '#35C759',
+	statusWarning: '#C93400',
+	statusCritical: '#000000',
+	statusActive: '#0879C4',
+	inputBackground: '#53565F',
+	inputColor: '#FFFFFF',
+	inputPlaceholder: '#C9CBD2',
+	inputDisabled: '#7D808A',
+	inputFocused: '#C9CBD2',
+	textDefault: '#FFFFFF',
+	textHelp: '#C9CBD2',
+	textInactive: '#7D808A'
+};
+
+function validateDomain() {
+	btnCaptureDomain.disabled = !domain.checkValidity();
+}
+
+function rgbToHex(r, g, b) {
+	const hexToReturn = [r, g, b]
+		.map((x) => {
+			const hex = x.toString(16);
+			return hex.length === 1 ? `0${hex}` : hex;
```

Comment:
this whole map could be `map(x => x.toString(16).padStart(2, '0'))`

Reply:
done